egui/src/widgets/text_edit/builder.rs

```diff
@@ -1250,10 +1308,32 @@ fn next_word_boundary_char_index(it: impl Iterator<Item = char>, mut index: usiz
     index
 }
 
+fn next_line_boundary_char_index(it: impl Iterator<Item = char>, mut index: usize) -> usize {
+    let mut it = it.skip(index);
+    if let Some(_first) = it.next() {
+        index += 1;
+
+        if let Some(second) = it.next() {
+            index += 1;
+            for next in it {
+                if is_line_char(next) != is_line_char(second) {
+                    break;
+                }
+                index += 1;
+            }
+        }
+    }
+    index
+}
+
 fn is_word_char(c: char) -> bool {
     c.is_ascii_alphanumeric() || c == '_'
 }
 
+fn is_line_char(c: char) -> bool {
```

Comment:
nitpick: "line character" is not a very established expression afaik. When I read `is_line_char(x)` I didn't understand it. I think it would make the code more readable to replace `is_line_char(c)` with `!is_linebreak(c)`

Reply:
Done